src/popup/popup.js

```diff
@@ -79,6 +79,7 @@ async function nextRequest() {
   console.log(url);
 
   const images = await fetchImages(url);
+  checkResultLength(images);
   addImagesToDOM(primaryGridMasonryObject, images, elements.gridPrimary);
   appObject.pageNo += 1;
 }
```

Comment:
This would fix the issue but we can do better:

- These 4 lines are exactly similar and analogous to calling `search(url)` function here. You can go ahead and replace these lines with the function call.

- Showing a notification to the user which says `No Images Found. Please enter a different query.` does not makes sense here because the user is not searching now, they are loading more search results by pressing the `Load more` button. It should say something like `No more images found.` Instead of creating a brand new function for this, you can modify `checkResultLength` itself to show different messages based on the context.

Reply:
@makkoncept can you please tell me how do I show two different ```showNotification()``` inside the same if condition. Can I implement all the functionalities of ```checkResultLength()``` inside ```nextRequest()``` like this? 
```
async function nextRequest() {
  let url;
  const { searchContext } = appObject;
  if (searchContext === 'collection') {
    url = getCollectionsUrl();
  } else if (searchContext === 'default') {
    url = getRequestUrl();
  } else if (searchContext === 'image-tag') {
    url = getTagsUrl();
  }
  console.log(url);

  const images = await fetchImages(url);
  if (images.length === 0) {
    showNotification(
      'No More Images Found.',
      'negative',
      'notification--extension-popup',
      4000,
    );
    removeSpinner(elements.spinnerPlaceholderPrimary);
    removeLoadMoreButton(elements.loadMoreSearchButtonWrapper);
    primaryGridMasonryObject.layout();
    throw new Error('No image found');
  } else {
    // render the "Load More" button if non empty result
    addLoadMoreButton(elements.loadMoreSearchButtonWrapper);
  }
  addImagesToDOM(primaryGridMasonryObject, images, elements.gridPrimary);
  appObject.pageNo += 1;
}
```

Reply:
Another good way would be to send another argument to `checkResultLenght()`, say `context`, which can have 2 strings: either `forSearch` or `forNextRequest`. 

``` javascript
if (context === 'forSearch') {
  message = 'No Images Found. Please enter a different query.'
}
else if (context === 'forNextRequest') {
  message = 'No more images found.';
}
showNotification(
      message,
      'negative',
      'notification--extension-popup',
      4000,
);    
```

Reply:
@makkoncept I made the changes, please have a look.